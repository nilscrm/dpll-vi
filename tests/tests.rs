use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
