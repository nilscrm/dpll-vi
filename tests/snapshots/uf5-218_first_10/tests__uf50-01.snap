---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = false
var 2 = true
var 3 = false
var 4 = true
var 5 = true
var 6 = true
var 7 = true
var 8 = true
var 9 = true
var 10 = false
var 11 = false
var 12 = true
var 13 = false
var 14 = true
var 15 = false
var 16 = false
var 17 = false
var 18 = false
var 19 = true
var 20 = true
var 21 = false
var 22 = false
var 23 = true
var 24 = false
var 25 = false
var 26 = false
var 27 = true
var 28 = false
var 29 = false
var 30 = false
var 31 = false
var 32 = false
var 33 = false
var 34 = false
var 35 = true
var 36 = true
var 37 = true
var 38 = false
var 39 = true
var 40 = false
var 41 = false
var 43 = false
var 44 = false
var 45 = false
var 46 = false
var 47 = false
var 48 = true
var 49 = true
var 50 = false

----- stderr -----

Interactions
  Total           24_078_149
    Depth         12_287_351
    Breadth                1
  Annihilate      13_582_858
  Commute             40_030
  Copy             2_324_256
  Erase            2_752_308
  Expand           1_902_435
  Call             2_203_114
  Branch           1_273_148

Memory
  Heap               520_592 B
  Allocated      533_897_584 B
  Freed          533_897_584 B

[Performance...]
